scarb/src/core/manifest/target.rs

```diff
@@ -83,6 +83,13 @@ impl Target {
         self.kind == TargetKind::CAIRO_PLUGIN
     }
 
+    pub fn is_builtin(&self) -> bool {
+        self.params
+            .get("builtin")
+            .and_then(|v| v.as_bool())
+            .unwrap_or(false)
+    }
+
```

Comment:
```rs
#[derive(Debug)]
#[non_exhaustive]
pub struct TargetInner {
    pub kind: TargetKind,
    pub name: SmolStr,
    pub source_path: Utf8PathBuf,
    pub group_id: Option<SmolStr>,
    pub params: toml::Value,
}
```

I assume it was intentional to have params as toml table, not sure how else we should parse it without fundamentally modifying `Target`

Reply:
> I assume it was intentional to have params as toml table

It is! It can hold any correct toml value. 


> not sure how else we should parse it without fundamentally modifying Target

It just does not need  to be part of `Target` implementation. The way you parse it is alright, but it should not be implemented in `Target` (i.e. please don't do `target.is_builtin()` but rather have `is_builtin(target) somewhere else).

Reply:
Addressed in [af8dc5b](https://github.com/software-mansion/scarb/pull/1605/commits/af8dc5b82b2a768f1be5967858841d9c7ca28875)